proposals/0000-shared-storage-property-wrapper.md

```diff
@@ -0,0 +1,403 @@
+shared let baz2$shared = Shared()
+
+var baz: WithShared<WithShared<Bool>> = WithShared(wrappedValue: WithShared(wrappedValue: false, baz$shared), baz2$shared)
+```
+
+## Impact on existing code
+
+This is an additive feature, and it shouldn't impact existing source code. 
+
+### Backward compatibility
+
+However, from a library evolution standpoint, making an existing property wrapper opt-in into the shared storage model can be a non-resilient change for ABI-public property wrappers. 
+
+Consider a type that exposes a property with a property wrapper to its public API. 
+
+```swift
+@propertyWrapper
+public struct Wrapper<Value> {
+  var wrappedValue: Value { ... }
+  var projectedValue: Wrapper { ... } 
+}
+
+public struct Container {
+  @Wrapper public var someValue: String
+}
+
+// -------
+// the generated interface
+public struct Container {
+  public var someValue: String 
+  public var $someValue: Wrapper 
+}
+
+@propertyWrapper
+public struct Wrapper<Value> { ... }
+```
+
+Suppose that on a later version, the author of this property wrapper decides to change it by adding shared storage. Even if the shared storage is given a default argument in the property wrapper initializer, this is a non-resilient change. The same would be true for the opposite scenario: removing the shared storage from an ABI-public wrapper.
+
+The example shows an API-level property wrapper, but the same would apply to an ABI-public implementation detail wrapper.  
+
+### Alternatives considered
+
+**Static shared storage**
+
+Instead of introducing a new attribute, we could store the generated storage property as a normal static variable in its enclosing instance.
+
+```swift
+@propertyWrapper(shared: Storage)
+struct Clamped {
+  private var value: Value
+  
+  var wrappedValue: Value { fatalError("use the subscript!") }
+  
+  // wrappedValue would be accessed through a subscript
+  subscript(shared storage: Storage) -> Value {
+    get { value }
+    set { value = storage.range.clamping(newValue) }
+  }
+  
+  struct Storage {
+    let range: ClosedRange<Value>
+    init(range: ClosedRange<Value>) { // ... } 
+  }
+} 
+
+// .... using it
+
+struct Hud {
+  @Clamped(range: 0...100) var value = 100
+}
+
+// Desugared version:
+struct Hud {
+  private var _value: Clamped<Int> = .init(wrappedValue: 100)
+  private static let _value$shared: Clamped<Int>.Storage = .init(range: 0...100)
+  var value: Int {
+    get { _value[shared: Hud._value$shared] }
+    set { _value[shared: Hud._value$shared] = newValue }
+  }
+}
+```
+
+Readability is one of the main disadvantages of this approach, as it would require passing the storage around through subscripts. 
+
+### Related Work
+
+The Future Directions sections on both SE-0258 and SE-0293, and the threads discussing this feature, especially this [post](https://www.notion.so/Copy-of-Shared-storage-for-Property-Wrappers-GSoC-Proposal-c6ac301b7e6a401e960a5e2a06adf962), were essential to this proposal.
```

Comment:
This notion link looks wrong. Is this supposed to be linking to https://forums.swift.org/t/future-directions-of-property-wrappers/27934 (or this specific comment: https://forums.swift.org/t/future-directions-of-property-wrappers/27934/11) or something else?